src/AutoRest.CSharp/Common/Input/CadlInputConstantConverter.cs

```diff
@@ -0,0 +1,105 @@
+        {
+            var isFirstProperty = id == null;
+            Object? value = null;
+            InputType? type = null;
+
+            reader.TryReadReferenceId(ref isFirstProperty, ref id);
+            if (!reader.TryReadWithConverter(nameof(InputConstant.Type), options, ref type))
+            {
+                throw new JsonException("Must provide type ahead of value.");
+            }
+            value = ReadConstantValue(ref reader, nameof(InputConstant.Value), options, type);
+
+            type = type ?? throw new JsonException("InputConstant must have type");
+
+            value = value ?? throw new JsonException("InputConstant must have value");
+
+            var constant = new InputConstant(value, type);
+            if (id != null)
+            {
+                resolver.AddReference(id, constant);
+            }
+            return constant;
+        }
+
+        public static object? ReadConstantValue(ref Utf8JsonReader reader, string propertyName, JsonSerializerOptions options, InputType? type)
+        {
+            if (type == null)
+            {
+                throw new JsonException("Must place type ahead of value.");
+            }
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException();
+            }
+
+            if (reader.GetString() != propertyName)
+            {
+                throw new JsonException("This is not for json field " + propertyName);
+            }
+
+            reader.Read();
+            Object? value = null;
+            switch (type) {
+                case InputPrimitiveType primitype:
+                    switch (primitype.Kind)
+                    {
+                        case InputTypeKind.String:
+                            value = reader.GetString() ?? throw new JsonException();
+                            break;
+                        case InputTypeKind.Uri:
+                            var stringvalue = reader.GetString() ?? throw new JsonException();
+                            value = new Uri(stringvalue);
+                            break;
+                        case InputTypeKind.Int32:
+                            value = reader.GetInt32();
+                            break;
+                        case InputTypeKind.Int64:
+                            value = reader.GetInt64();
+                            break;
+                        case InputTypeKind.Boolean:
+                            value = reader.GetBoolean();
+                            break;
+                        default:
+                            value = reader.GetString() ?? throw new JsonException();
+                            break;
+
+                    }
+                    break;
+                case InputModelType model:
+                    throw new JsonException("Not supported type.");
+                default:
+                    break;
```

Comment:
Let's throw exception for `default` as well. That way we should be able to return `object` instead of `object?`

Reply:
Thanks. I will update it.